lib/groupify/adapter/active_record/association_extensions.rb

```diff
@@ -0,0 +1,100 @@
+module Groupify
+  module ActiveRecord
+    module AssociationExtensions
+      extend ActiveSupport::Concern
+
+      # Defined to create alias methods before
+      # the association is extended with this module
+      def <<(*)
+        super
+      end
+
+      def add_without_exception(*children)
+        add_children_to_parent(children.flatten, false)
+      end
+
+      def add_with_exception(*children)
+        add_children_to_parent(children.flatten, true)
+      end
+
+      alias_method :add_as_usual, :<<
+      alias_method :<<, :add_without_exception
+      alias_method :add, :add_with_exception
+
+      def as(membership_type)
+        return self unless membership_type
+        where(group_memberships: {membership_type: membership_type})
```

Comment:
Per https://github.com/dwbutler/groupify/pull/58 this needs to be changed to `merge(Groupify.group_membership_klass.as(membership_type))` to support a custom group membership model.